test/lib/jdk/test/lib/util/CoreUtils.java

```diff
@@ -148,6 +150,11 @@ public static String getCoreFileLocation(String crashOutputString, long pid) thr
                     // We can't generate cores files with hardened binaries on OSX 10.15 and later.
                     throw new SkippedException("Cannot produce core file with hardened binary on OSX 10.15 and later");
                 }
+            } else {
```

Comment:
Maybe you could do just one case here:
`else if (!Platform.hasPlistEntriesOSX()) {`...